skillmap/src/components/AppModal.tsx

```diff
@@ -129,6 +135,39 @@ export class AppModalImpl extends React.Component<AppModalProps> {
             <textarea className="report-abuse-text" placeholder={abuseModalText} />
         </Modal>
     }
+
+    renderCodeCarryoverModal() {
+        const  { dispatchHideModal, skillMap, activity, pageSourceUrl, userState, dispatchSetReloadHeaderState } = this.props;
+        const carryoverModalTitle = lf("Keep code from previous activity?");
+        const carryoverModalText = lf("Do you want to start with your code from the previous activity or start fresh with starter code? Your images, tilemaps, tiles, and animations will stick around either way.");
+
+        const actions = [
+            { label: lf("START FRESH"), onClick: async () => {
+                tickEvent("skillmap.startfresh");
+
+                dispatchSetReloadHeaderState("reloading")
+
+                await carryoverProjectCode(userState!, pageSourceUrl!, skillMap!, activity!.activityId, false)
+
+                dispatchSetReloadHeaderState("reload");
```

Comment:
should it be "reload" in both start fresh/carryover cases? we don't need to reload if they select the default option, right?

Reply:
we do actually, because i merge the user's project into the currently loaded one.

Reply:
hmm, feasible/worth it to kick off the "start fresh" merge as soon as the project is loaded? i don't know how long it's really taking--maybe it's fast?

Reply:
i think it's pretty fast currently. The first editor load is what takes the most time